tests/cli/main_tool.py

```diff
@@ -129,11 +133,22 @@ def testRecipeSetupArgs(self, recipe_name):
 
   @parameterized.named_parameters(_EnumerateRecipeNames())
   def testRecipeValidators(self, recipe_name):
-    """Tests that recipes do not specify invalid validators."""
+    """Parameterised version of _testRecipeValidators."""
+    self._testRecipeValidators(recipe_name)
+
+  def _testRecipeValidators(self, recipe_name):
+    """Tests that a recipe does not specify invalid validators."""
     # pylint: disable=protected-access
     self.tool._state = dftw_state.DFTimewolfState(config.Config)
     recipe = self.tool._recipes_manager.Recipes()[recipe_name]
 
+    test_params = recipe.GetTestParams()
+    if test_params:
+      recipe_args = [recipe_name] + test_params
+      self.tool.ParseArguments(recipe_args)
+    else:
+      self.fail('No test_params in recipe')
```

Comment:
This essentially makes the test_params mandatory for all recipes, even those that may not have any (as unlikely a it sounds it is something that is supported). I'm not against it, but wanted to make sure that was on your radar.

Reply:
Having `test_params` required for every recipe was indeed the intent, but the no args edge case didn't occur to me. Updated, with test cases.